### PR TITLE
Replace Macros with Aliases

### LINKED ISSUE #117: Universal Bindings
We are [running into some issues while generating cross platform bindings for .NET](https://github.com/Speykious/NativeFileDialogs.NET/pull/3#pullrequestreview-1773626280). It mostly arises from the use of macros to define the utf8 functions on Mac and Linux. On Windows, both native and utf8 functions get exported, but on Mac and Linux, only native functions get exported.

I see two potential solutions:
* On Mac and Linux, change the utf8 macros to function definitions, so that all 3 platforms have both sets of function exports.
* On Mac and Linux, change utf8 to be the default and native to be the macro.

### PR DESCRIPTION
**Pull request by ds5678**
This pull request is intended to address issues with creating wrappers in other languages. I made Utf8 function exports available on all platforms.

Supercedes #118 

Resolves #117

### CODE REVIEW

**Comment by btzy** [comment]
Thanks!  This looks okay in principle, but I think I'll want to add a CI build configuration to build with the `NFD_NATIVE` define, which currently isn't tested in CI.  (I'll add this in a separate PR first, and then rebase this PR over that one.)

**Comment by btzy** [comment]
Hi, can you rebase this PR over the latest master branch please?

**Comment by ds5678** [comment]
All good?

File: src/include/nfd.h [hunk 1]
```diff
@@ -279,16 +284,6 @@ typedef nfdnfilteritem_t nfdfilteritem_t;
 #define NFD_PathSet_GetPath NFD_PathSet_GetPathN
 #define NFD_PathSet_FreePath NFD_PathSet_FreePathN
 #define NFD_PathSet_EnumNext NFD_PathSet_EnumNextN
```
Comment (btzy) [inline src/include/nfd.h:286]: Could we make the defines from line 277 to 286 of the new file not depend on `_WIN32` please?  This won't change the behaviour of calling these macros, but will reduce unnecessary platform-specific differences.

i.e. something like:
```
#ifdef _WIN32

/* say that the U8 versions of functions are not just __attribute__((alias(""))) to the native
 * versions */
#define NFD_DIFFERENT_NATIVE_FUNCTIONS

#endif

#ifdef NFD_NATIVE
typedef nfdnchar_t nfdchar_t;
typedef nfdnfilteritem_t nfdfilteritem_t;
#define NFD_FreePath NFD_FreePathN
#define NFD_OpenDialog NFD_OpenDialogN
#define NFD_OpenDialogMultiple NFD_OpenDialogMultipleN
#define NFD_SaveDialog NFD_SaveDialogN
#define NFD_PickFolder NFD_PickFolderN
#define NFD_PathSet_GetPath NFD_PathSet_GetPathN
#define NFD_PathSet_FreePath NFD_PathSet_FreePathN
#define NFD_PathSet_EnumNext NFD_PathSet_EnumNextN
#else
typedef nfdu8char_t nfdchar_t;
typedef nfdu8filteritem_t nfdfilteritem_t;
#define NFD_FreePath NFD_FreePathU8
#define NFD_OpenDialog NFD_OpenDialogU8
#define NFD_OpenDialogMultiple NFD_OpenDialogMultipleU8
#define NFD_SaveDialog NFD_SaveDialogU8
#define NFD_PickFolder NFD_PickFolderU8
#define NFD_PathSet_GetPath NFD_PathSet_GetPathU8
#define NFD_PathSet_FreePath NFD_PathSet_FreePathU8
#define NFD_PathSet_EnumNext NFD_PathSet_EnumNextU8
#endif  // NFD_NATIVE
```

**Comment by btzy** [comment]
Thanks!  Just one minor comment, and I'll merge it once that is changed.  I suppose Clang/ObjC doesn't have attribute alias, but what you've done is fine as well, since it's just an implementation detail now.

**Comment by btzy** [comment]
Seems like NFD_PickFolderU8 is missing on nfd_gtk.cpp?

**Review by btzy — APPROVED** [review]
Thanks!

**Pull request merged**